pages/search.js

```diff
@@ -1,179 +1,41 @@
 import PropTypes from "prop-types";
-import { useState } from "react";
-import Router from "next/router";
+import { API_URL } from "../lib/constants";
+import { getQueryStringFromServerQuery } from "../lib/query-utils";
 import Breadcrumbs from "../components/breadcrumbs";
-import { DataPanel, DataItemLabel } from "../components/data-area";
 import JsonDisplay from "../components/json-display";
 import PagePreamble from "../components/page-preamble";
-import RegulomeVersionTag from "../components/regulome-version-tag";
-import { API_URL } from "../lib/constants";
-import { getQueryStringFromServerQuery } from "../lib/query-utils";
-
-const inputClassName =
-  "bg-gray-200 appearance-none border-2 border-gray-200 rounded w-full py-2 px-4 text-gray-700 leading-tight focus:outline-none focus:bg-white focus:border-brand";
-const buttonClassName =
-  "shadow bg-brand focus:shadow-outline focus:outline-none text-white font-bold py-2 px-4 rounded";
-
-export default function Page({ data = null }) {
-  const [fileInput, setFileInput] = useState("");
-
-  // Handles the submit event on form submit.
-  async function handleSubmit(event) {
-    // Stop the form from submitting and refreshing the page.
-    event.preventDefault();
-
-    // Get data from the form.
-    const textInput = event.target.regions.value;
-    const assembly = event.target.assembly.value;
-    const maf = event.target.maf.value;
-
-    if (textInput && fileInput) {
-      alert(
-        "Only choose one method to set regions: throughtext input or file upload"
-      );
-    } else if (!textInput && !fileInput) {
-      alert("Please define region(s) for query");
-    } else {
-      const regions = textInput
-        ? textInput.replace(/\s/g, " ")
-        : fileInput.replace(/\s/g, " ");
-      const query =
-        maf === "1.1"
-          ? {
-              regions,
-              genome: assembly,
-            }
-          : {
-              regions,
-              genome: assembly,
-              maf,
-            };
-      Router.push({
-        pathname: "/search",
-        query,
-      });
-    }
-  }
-
-  async function readText(event) {
-    const file = event.target.files.item(0);
-    const text = await file.text();
-    setFileInput(text);
-  }
 
+export default function Page({ data }) {
```

Comment:
Naming all your page components `Page` is perfectly fine, but it does have a little impact on debugging which you might or might not feel is important. With a Javascript crash, the Chrome debugger shows the name of the function that crashed, which in your case will be `Page` no matter what page crashed. You can hover over the name of the file (e.g. `[id].js`) and it’ll show you the path which _will_ uniquely identify the location of the crash. So something to think about.

Reply:
Will rename the page compenents.